mod_ci/controllers.py

```diff
@@ -304,107 +304,46 @@ def kvm_processor(app, db, kvm_name, platform, repository, delay) -> None:
-    try:
-        repo = Repo(os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'vm_data', kvm_name, 'unsafe-ccextractor'))
-    except InvalidGitRepositoryError:
-        log.critical(f"[{platform}] Could not open CCExtractor's repository copy!")
-        return
-
-    # Return to master
-    repo.heads.master.checkout(True)
-    # Update repository from upstream
-    try:
-        github_url = test.fork.github
-        if is_main_repo(github_url):
-            origin = repo.remote('origin')
-        else:
-            fork_id = test.fork.id
-            remote = f'fork_{fork_id}'
-            if remote in [remote.name for remote in repo.remotes]:
-                origin = repo.remote(remote)
-            else:
-                origin = repo.create_remote(remote, url=github_url)
-    except ValueError:
-        log.critical(f"[{platform}] Origin remote doesn't exist!")
-        return
-
-    fetch_info = origin.fetch()
-    if len(fetch_info) == 0:
-        log.info(f'[{platform}] Fetch from remote returned no new data...')
-    # Checkout to Remote Master
-    repo.git.checkout(origin.refs.master)
-    # Pull code (finally)
-    pull_info = origin.pull('master')
-    if len(pull_info) == 0:
-        log.info(f"[{platform}] Pull from remote returned no new data...")
-    elif pull_info[0].flags > 128:
-        log.critical(f"[{platform}] Did not pull any information from remote: {pull_info[0].flags}!")
-        return
-
-    ci_branch = 'CI_Branch'
-    # Delete the test branch if it exists, and recreate
-    try:
-        repo.delete_head(ci_branch, force=True)
-    except GitCommandError:
-        log.info(f"[{platform}] Could not delete CI_Branch head")
-
-    # Remove possible left rebase-apply directory
-    try:
-        shutil.rmtree(os.path.join(config.get('SAMPLE_REPOSITORY', ''), 'unsafe-ccextractor', '.git', 'rebase-apply'))
-    except OSError:
-        log.info(f"[{platform}] Could not delete rebase-apply")
-    # If PR, merge, otherwise reset to commit
-    if test.test_type == TestType.pull_request:
-        # Fetch PR (stored under origin/pull/<id>/head)
-        pull_info = origin.fetch(f'pull/{test.pr_nr}/head:{ci_branch}')
-        if len(pull_info) == 0:
-            log.warn(f"[{platform}] Did not pull any information from remote PR!")
-        elif pull_info[0].flags > 128:
-            log.critical(f"[{platform}] Did not pull any information from remote PR: {pull_info[0].flags}!")
-            return
-
-        try:
-            test_branch = repo.heads[ci_branch]
-        except IndexError:
-            log.critical(f'{ci_branch} does not exist')
-            return
-
-        test_branch.checkout(True)
-
-        try:
-            pull = repository.pulls(f'{test.pr_nr}').get()
-        except ApiError as a:
-            log.error(f'Got an exception while fetching the PR payload! Message: {a.message}')
-            return
-        if pull['mergeable'] is False:
```

Comment:
w.r.t this, since if there is a merge conflict builds are not triggered (example - https://github.com/Tarun-Arora/ccextractor/pull/21), hence no tests would run, so we no longer need to check this.